continuous-integration/reference-architectures/README.md

```diff
@@ -1,3 +1,181 @@
+or [Travis CI](https://www.travis-ci.com/), serve as management node for CI operations.
+3. _Continuous Deployment System_  
+Tooling and services, such as [Ansible](https://www.ansible.com/) or [Puppet](https://puppet.com/), 
+that deploy published software into one or more venues for extended testing or delivery. Often, 
+software outputs from the CI pipeline are deployed to development systems for extended user testing.
+
+### Build systems are composed of discrete process steps
+The compile-release-publish workflow is broken down to a series of discrete process steps that describe 
+specific operations, ranging from commit processing to publishing compiled binaries. These discrete 
+process steps are reported in logs and reports, helping identify specific points during processing 
+and build failures. Process steps are associated with specific systems in the pipeline based on its 
+role, so the Development System may focus on tracking code merging and commit history, for instance. 
+
+#### Development System
+- **Commit**  
+Code management, including developer commits, merging and history, are tracked in this step. A commit 
+triggers notifications to subsequent steps and systems to kick off the CI pipeline. 
+- **Validate** (optional)  
+Self-contained processing may take place to validate commit content in advance of triggering 
+notifications to subsequent steps and systems. These may include facilities, similar to 
+[Github Actions](https://docs.github.com/en/actions), that can be leveraged to validate files, amongst 
+other tasks. 
+
+#### Continuous Integration System
+- **Compile** (core)  
+Core software packages are compiled and the capacity to compile is validated, resulting in unpackaged 
+binaries. Failures at this step indicate a problem in code, code merges or tooling. 
+- **Test**   
+At the core of CI is testing. The process is inherently checking for successful compilation and 
+execution, and optionally validating software with unit, integration and other tests.  
+_Successful Execution:_ Builds are inherently validated to ensure compilations complete successfully, 
+and other processes germane to the build complete. **Errors or failures result in build failure where 
+the compile-release-publish process is halted.**  
+_Software Testing:_ (optional, desirable) Test software packages are compiled and the capacity to compile tests is validated, also. The test 
+step will execute the compiled tests against core software packages to run [unit and other types of 
+functional tests](https://nasa-ammos.github.io/slim/continuous-testing/reference-architectures/) on 
+every build. Builds will proceed if no tests are implemented, however, test validation is material 
+to the CI philosophy.  
+- **Report**  
+At the core of CI is reporting, both notifications and functional reporting.  
+_Notifications:_ Developers are notified on successful completion of builds or failures. 
```

Comment:
Suggest adding some examples of available notification methods like email, chat services(i.e. slack, hipchat), SMS text, AWS SNS, etc.

Reply:
Added.